ui/analyse/src/ctrl.ts

```diff
@@ -353,7 +353,8 @@ export default class AnalyseCtrl {
       config.movable!.color = color;
     }
     config.premovable = {
-      enabled: config.movable!.color && config.turnColor !== config.movable!.color,
+      enabled:
+        config.movable!.color && config.turnColor !== config.movable!.color && !this.currPosition().isEnd(),
```

Comment:
what's the point of analysis premoves again? Since we can move all pieces.

I suppose it's only useful in specific applications like "learn from your mistakes" where only one side is playable. That would be a better thing to check.

Reply:
@ornicar I think the main point is when the connection to the server is poor. E.g., if you open a study board and turn off wifi, the next move should be a premove.